openpype/plugins/publish/extract_slate_global.py

```diff
@@ -0,0 +1,541 @@
+import os
+import re
+import logging
+import tempfile
+
+import opentimelineio as otio
+import pyblish.api
+from html2image import Html2Image
+
+from openpype.lib import (
+    get_oiio_tools_path,
+    get_ffprobe_streams,
+    get_chrome_tool_path,
+    run_subprocess,
+)
+from openpype.pipeline import publish
+
+
+def timecode_to_frames(timecode, framerate):
+    rt = otio.opentime.from_timecode(timecode, framerate)
+    return int(otio.opentime.to_frames(rt))
+
+
+def frames_to_timecode(frames, framerate):
+    rt = otio.opentime.from_frames(frames, framerate)
+    return otio.opentime.to_timecode(rt)
+
+
+def frames_to_seconds(frames, framerate):
+    rt = otio.opentime.from_frames(frames, framerate)
+    return otio.opentime.to_seconds(rt)
+
+
+class SlateCreator:
+    """Class that formats and renders an html template to use as a slate.
+
+    Html renders as sRGB (or Rec.709 if you want a little gamma boost), so
+    rendering will happen in .png as intermediate file, while then converted
+    to dest cspace using oiio --colorconvert attributes syntax.
+
+    """
+
+    def __init__(
+        self,
+        template_path,
+        resources_path="",
+        staging_dir="",
+        staging_subfolder="",
+        data=None,
+    ):
+        self.data = data.copy() if data else {}
+        self.log = logging.getLogger(self.__class__.__name__)
```

Comment:
This should get logger from plugin probably, otherwise the logs won't be available in UI.

Reply:
Or just pass it in arguments?

Reply:
done!